src/app/ubs/ubs-admin/components/ubs-admin-table/table-cell-input/table-cell-input.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { IAlertInfo, IEditCell } from '@ubs/ubs-admin/models/edit-cell.model';
+import { IColumnBelonging } from '@ubs/ubs-admin/models/ubs-admin.interface';
+import { AdminTableService } from '@ubs/ubs-admin/services/admin-table.service';
+import { take } from 'rxjs';
+import { CommentPopUpComponent } from '../../shared/components/comment-pop-up/comment-pop-up.component';
+import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
+import { LocalStorageService } from '@global-service/localstorage/local-storage.service';
+@Component({
+  selector: 'app-table-cell-input',
+  templateUrl: './table-cell-input.component.html',
+  styleUrls: ['./table-cell-input.component.scss']
+})
+export class TableCellInputComponent {
+  @Input() column: IColumnBelonging;
+  @Input() id: number;
+  @Input() ordersToChange: number[];
+  @Input() isAllChecked: boolean;
+  @Input() isUneditableStatus: boolean;
+  @Input() data: string;
+
+  @Output() cancelEdit = new EventEmitter();
+  @Output() editCommentCell = new EventEmitter();
+  @Output() showBlockedInfo = new EventEmitter();
+
+  isEditable: boolean;
+  isBlocked: boolean;
+
+  private typeOfChange: number[];
+  private font = '12px Lato, sans-serif';
+
+  private dialogConfig = new MatDialogConfig();
+
+  constructor(
+    private adminTableService: AdminTableService,
+    private localStorageService: LocalStorageService,
+    public dialog: MatDialog
+  ) {}
+
+  edit(): void {
+    this.isEditable = false;
+    this.isBlocked = true;
+
+    this.typeOfChange = this.adminTableService.howChangeCell(this.isAllChecked, this.ordersToChange, this.id);
+    this.adminTableService
+      .blockOrders(this.typeOfChange)
+      .pipe(take(1))
+      .subscribe((res: IAlertInfo[]) => {
+        if (res[0] === undefined) {
```

Comment:
We can flip condition:
```
if (res && res[0]) 
```
in both ```this.isBlocked = false;```

Handle any errors that might occur during the request